packages/jsii-rosetta/lib/commands/transliterate.ts

```diff
@@ -0,0 +1,172 @@
+  }
+  const assemblies = await Promise.all(
+    assemblyLocations.map(async (assemblyLocation) => {
+      const assembly = (await readJson(
+        resolve(assemblyLocation, SPEC_FILE_NAME),
+      )) as Assembly;
+      await rosetta.addAssembly(assembly, assemblyLocation);
+      return { assemblyLocation, assembly };
+    }),
+  ).then((items) =>
+    items.reduce((acc, { assemblyLocation, assembly }) => {
+      acc.set(assemblyLocation, assembly);
+      return acc;
+    }, new Map<string, Assembly>()),
+  );
+
+  for (const [location, assembly] of assemblies.entries()) {
+    console.error(
+      `Working on the assembly ${assembly.name}@${assembly.version} at ${location}`,
+    );
+    for (const language of targetLanguages) {
+      const now = new Date().getTime();
+      console.error(`- Transliterating to ${language}`);
+      const result = mutableCopy(assembly);
+      if (result.readme?.markdown) {
+        result.readme.markdown = rosetta.translateSnippetsInMarkdown(
+          result.readme.markdown,
+          language,
+          options.strict ?? false,
+          (translation) => ({
+            language: translation.language,
+            source: prefixDisclaimer(translation),
+          }),
+        );
+      }
+      for (const type of Object.values(result.types ?? {})) {
+        transliterateType(type, rosetta, language, options.strict ?? false);
+      }
+      // eslint-disable-next-line no-await-in-loop
+      await writeJson(
+        resolve(location, `${SPEC_FILE_NAME}.${language}`),
+        result,
+      );
+      const then = new Date().getTime();
+      console.error(`  => Done after ${then - now} milliseconds`);
+    }
+  }
+}
+
+type Mutable<T> = T extends ReadonlyArray<infer E>
+  ? Array<Mutable<E>>
+  : T extends Array<infer E>
+  ? Array<Mutable<E>>
+  : T extends string
+  ? T
+  : { -readonly [K in keyof T]: Mutable<T[K]> };
+function mutableCopy<T>(val: T): Mutable<T> {
+  if (val == null || typeof val !== 'object') {
+    return val as any;
+  }
+  if (Array.isArray(val)) {
+    return val.map(mutableCopy) as any;
+  }
+  const result = {} as Mutable<T>;
+  for (const [key, value] of Object.entries(val)) {
+    (result as any)[key] = mutableCopy(value);
+  }
+  return result;
+}
+
+function prefixDisclaimer(translation: Translation): string {
+  const message = translation.didCompile
+    ? 'Example automatically generated. See https://github.com/aws/jsii/issues/826'
+    : 'Example automatically generated without compilation. See https://github.com/aws/jsii/issues/826';
+  return `${commentToken()} ${message}\n${translation.source}`;
+
+  function commentToken() {
+    // This is future-proofed a bit, but don't read too much in this...
+    switch (translation.language) {
+      case 'python':
+      case 'ruby':
+        return '#';
+      case 'csharp':
+      case 'java':
+      case 'go':
+      default:
+        return '//';
+    }
+  }
+}
+
+function transliterateType(
```

Comment:
I am surprised this code doesn't exist already somewhere...

Reply:
Rosetta just didn't get in the business of final result substitutions previously... This particular feature is scattered in consumers of Rosetta tablet files... Which is parts of what I am trying to solve here.